### PR TITLE
Release 2.2.0

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] Release 2.2.0
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,42 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 2.2.0 - TioDeNadal [2024-08-23]
```
Comment (Lucpen) [inline CHANGELOG.md:6]:
```suggestion
## 2.2.0 - TioDeNadal [2024-08-26]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,42 @@
...
+### `Parameters`
+
+| Old parameter | New parameter                  |
+| ------------- | ------------------------------ |
+|               | `--gencode_annotation_version` |
+|               | `--vep_refs_download`          |
+|               | `--skip_download_vep`          |
+|               | `--skip_download_gnomad`       |
+
+:::note Parameter has been updated if both old and new parameter information is present. Parameter has been added if just thenew parameter information is present. Parameter has been removed if new parameter information isn't present. :::
```
Comment (fellen31) [inline CHANGELOG.md:27]: `:::note` is not formatting the text correctly on GitHub. 

```suggestion
> [!NOTE]
> Parameter has been updated if both old and new parameter information is present.
> Parameter has been added if just thenew parameter information is present.
> Parameter has been removed if new parameter information isn't present.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,42 @@
...
+
+### `Changed`
+
+- Updated modules bcftools/annotate, bcftools/mpileup, bcftools/view, cat/fastq, ensemblvep/filtervep, fastp, fastqc, gatk4/haplotypecaller, gatk4/splitncigarreads, gunzip, multiqc, picard/collectrnaseqmetrics, samtools/index, star/align, star/genomegenerate, stringtie/stringtie, tabix/bgziptabix, tabix/tabix/ and untar [#153](https://github.com/genomic-medicine-sweden/tomte/pull/153)
```
Comment (fellen31) [inline CHANGELOG.md:31]:
```suggestion
- Updated modules bcftools/annotate, bcftools/mpileup, bcftools/view, cat/fastq, ensemblvep/filtervep, fastp, fastqc, gatk4/haplotypecaller, gatk4/splitncigarreads, gunzip, multiqc, picard/collectrnaseqmetrics, samtools/index, star/align, star/genomegenerate, stringtie/stringtie, tabix/bgziptabix, tabix/tabix and untar [#153](https://github.com/genomic-medicine-sweden/tomte/pull/153)
```

File: docs/usage.md [hunk 4]
```diff
@@ -191,16 +196,17 @@ The mandatory and optional parameters for each category are tabulated below.
...
+| Mandatory | Optional                      |
+| --------- | ----------------------------- |
+|           | skip_vep<sup>1</sup>          |
+|           | vep_plugin_files<sup>2</sup>  |
+|           | vep_cache<sup>2</sup>         |
+|           | vep_cache_version<sup>3</sup> |
+|           | gene_panel_clinical_filter    |
+
+<sup>1</sup> If it is not provided by the user, the default value is false<br />
+<sup>2</sup> VEP cache and plugins can be automatically downloaded by the pipeline by setting `--skip_download_vep false`, `--skip_download_gnomad false` and providing a lcsv with a list of files to download `--vep_refs_download` as done [here](https://github.com/genomic-medicine-sweden/tomte/blob/dev/test_data/vep_to_download.csv). VEP caches can also be downloaded [here](https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#cache). VEP plugins may also be installed in the cache directory, and the plugin pLI is mandatory to install. To supply files required by VEP plugins, use `vep_plugin_files` parameter. See example cache [here](https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/vep_cache_and_plugins.tar.gz).<br />
```
Comment (fellen31) [inline docs/usage.md:208]: Can't find `--skip_download_vep`, `--skip_download_gnomad` and `--vep_refs_download` in a table, not sure if they should be or not.

File: modules/local/build_vep_cache.nf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        wget: \$(echo wget -V 2>&1 | grep "GNU Wget" | cut -d" " -f3 > versions.yml)
+    END_VERSIONS
+    """
+
+    stub:
+    """
+    touch vep_cache
```
Comment (fellen31) [inline modules/local/build_vep_cache.nf:51]: is `touch` ok or should it be `mkdir`?

File: modules/local/vep_gnomad_download.nf [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    if [[ "${genome}" == *"38"* ]]; then
+        bcftools concat gnomad.genomes.*${gnomad_version2download}.sites.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y}.vcf.bgz | bcftools annotate --output-type z --output gnomad_v${gnomad_version2download}.vcf.gz --include 'FILTER="PASS"' --remove ^INFO/AF,INFO/AF_grpmax,INFO/AF_popmax --threads $task.cpus --write-index=tbi
+    else
+        bcftools annotate --output-type z --output gnomad_v${gnomad_version2download}.vcf.gz --include 'FILTER="PASS"' --remove ^INFO/AF,INFO/AF_grpmax,INFO/AF_popmax --threads $task.cpus --write-index=tbi gnomad.genomes.*${gnomad_version2download}.sites.vcf.bgz
+    fi
+    rm gnomad.genomes.*${gnomad_version2download}.*.vcf.bgz*
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        wget: \$(echo wget -V 2>&1 | grep "GNU Wget" | cut -d" " -f3 > versions.yml)
```
Comment (fellen31) [inline modules/local/vep_gnomad_download.nf:45]: bcftools and tabix added in stub but not here.

File: modules/local/wget_download.nf [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gnu-wget:1.18--h5bf99c6_5' :
+        'quay.io/biocontainers/gnu-wget:1.18--h5bf99c6_5' }"
+
+    input:
+    tuple val(meta), val(path_wget)
+
+    output:
+    path(meta), emit: downloaded_file
+    path "versions.yml"  , emit: versions
```
Comment (fellen31) [inline modules/local/wget_download.nf:15]:
```suggestion
    path "versions.yml", emit: versions
```

File: nextflow_schema.json [hunk 8]
```diff
@@ -200,13 +228,11 @@
                 },
                 "skip_subsample_region": {
                     "type": "boolean",
-                    "default": false,
```
Comment (fellen31) [inline nextflow_schema.json:203]: Why remove the default here and below? 

I had problems with the linting before if that is the case, seems to be resolved now.

File: subworkflows/local/download_references.nf [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        // Read and store paths in vep_refs_download_unprocessed
+        ch_vep_refs_download_unprocessed.splitCsv(header: true)
+            .map { row -> return tuple(row.name, row.path_for_wget) }
+            .set { ch_vep_refs_download }
+
+        // Download files
+        WGET_DOWNLOAD(ch_vep_refs_download.filter{ it != null })
+        VEP_GNOMAD_DOWNLOAD(ch_genome, ch_vep_cache_version)
+
+        BUILD_VEP_CACHE(WGET_DOWNLOAD.out.downloaded_file.collect(),VEP_GNOMAD_DOWNLOAD.out.gnomad_vcf_tbi.flatten().collect())
```
Comment (fellen31) [inline subworkflows/local/download_references.nf:36]:
```suggestion
        BUILD_VEP_CACHE(WGET_DOWNLOAD.out.downloaded_file.collect(), VEP_GNOMAD_DOWNLOAD.out.gnomad_vcf_tbi.flatten().collect())
```

**Review by fellen31 — CHANGES_REQUESTED** [review]
Nice work!

File: modules/local/build_vep_cache.nf (hunk 5 above)
Comment (Lucpen) [inline modules/local/build_vep_cache.nf:51]:
```suggestion
    mkdir vep_cache
```

Comment (Lucpen) [reply modules/local/build_vep_cache.nf:51]: Good catch!

File: modules/local/vep_gnomad_download.nf [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    END_VERSIONS
```
Comment (Lucpen) [inline modules/local/vep_gnomad_download.nf:46]:
```suggestion
        wget: \$(echo wget -V 2>&1 | grep "GNU Wget" | cut -d" " -f3 > versions.yml)
         bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
         tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
    END_VERSIONS
```

File: modules/local/vep_gnomad_download.nf (hunk 6 above)
Comment (Lucpen) [reply modules/local/vep_gnomad_download.nf:45]: That's true, thanks!

File: docs/usage.md [hunk 11]
```diff
@@ -191,16 +196,17 @@ The mandatory and optional parameters for each category are tabulated below.
...
+<sup>3</sup> If it is not provided by the user, the default value is 110, supported values are 107 and 110
```
Comment (Lucpen) [inline docs/usage.md:209]:
```suggestion
| Mandatory | Optional                         |
| --------- | -------------------------------- |
|           | skip_vep<sup>1</sup>             |
|           | vep_plugin_files<sup>2</sup>     |
|           | vep_cache<sup>2</sup>            |
|           | vep_cache_version<sup>3</sup>    |
|           | skip_download_vep<sup>4</sup>    |
|           | skip_download_gnomad<sup>4</sup> |
|           | vep_refs_download                |
|           | gene_panel_clinical_filter       |

<sup>1</sup> If it is not provided by the user, the default value is false<br />
<sup>2</sup> VEP cache and plugins can be automatically downloaded by the pipeline by setting `--skip_download_vep false`, `--skip_download_gnomad false` and providing a lcsv with a list of files to download `--vep_refs_download` as done [here](https://github.com/genomic-medicine-sweden/tomte/blob/dev/test_data/vep_to_download.csv). VEP caches can also be downloaded [here](https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#cache). VEP plugins may also be installed in the cache directory, and the plugin pLI is mandatory to install. To supply files required by VEP plugins, use `vep_plugin_files` parameter. See example cache [here](https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/vep_cache_and_plugins.tar.gz).<br />
<sup>3</sup> If it is not provided by the user, the default value is 110, supported values are 107 and 110
<sup>4</sup> If it is not provided by the user, the default value true
```

File: docs/usage.md (hunk 4 above)
Comment (Lucpen) [reply docs/usage.md:208]: To be honest, I was doubting whether to add it or not, but it is likely better to have too much too much than too little 😄

File: modules/local/vep_gnomad_download.nf [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    else
+        bcftools annotate --output-type z --output gnomad_v${gnomad_version2download}.vcf.gz --include 'FILTER="PASS"' --remove ^INFO/AF,INFO/AF_grpmax,INFO/AF_popmax --threads $task.cpus --write-index=tbi gnomad.genomes.*${gnomad_version2download}.sites.vcf.bgz
+    fi
+    rm gnomad.genomes.*${gnomad_version2download}.*.vcf.bgz*
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        wget: \$(echo wget -V 2>&1 | grep "GNU Wget" | cut -d" " -f3 > versions.yml)
+         bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
+         tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
```
Comment (Lucpen) [inline modules/local/vep_gnomad_download.nf:47]:
```suggestion
        bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
        tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
```

File: nextflow_schema.json [hunk 13]
```diff
@@ -200,13 +228,11 @@
...
                     "description": "Turn off subsampling of the region. The region is defined by the subsample_bed parameter and the fraction is given by seed_frac",
                     "fa_icon": "fas fa-toggle-off"
                 },
                 "skip_downsample": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:235]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 14]
```diff
@@ -255,31 +281,26 @@
                 },
                 "skip_build_tracks": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:283]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 15]
```diff
@@ -255,31 +281,26 @@
...
-                    "default": false,
                     "description": "Skip building splice junction tracks for IGV.",
                     "fa_icon": "fas fa-toggle-off"
                 },
                 "skip_stringtie": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:288]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 16]
```diff
@@ -255,31 +281,26 @@
...
-                    "default": false,
                     "description": "Skip analysis with StringTie",
                     "fa_icon": "fas fa-toggle-off"
                 },
                 "skip_vep": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:293]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 17]
```diff
@@ -255,31 +281,26 @@
...
-                    "default": false,
                     "description": "Skip Ensembl Variant Effect Predictor",
                     "fa_icon": "fas fa-toggle-off"
                 },
                 "skip_drop_ae": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:298]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 18]
```diff
@@ -255,31 +281,26 @@
...
-                    "default": false,
                     "description": "Skip DROP Aberrant Expression module ",
                     "fa_icon": "fas fa-toggle-off"
                 },
                 "skip_drop_as": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:303]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json [hunk 19]
```diff
@@ -200,7 +228,6 @@
                 },
                 "skip_subsample_region": {
                     "type": "boolean",
```
Comment (Lucpen) [inline nextflow_schema.json:230]:
```suggestion
                    "type": "boolean",
                    "default": false,
```

File: nextflow_schema.json (hunk 8 above)
Comment (Lucpen) [reply nextflow_schema.json]: Done!

**Review by fellen31 — APPROVED** [review]

File: CHANGELOG.md [hunk 20]
```diff
@@ -3,6 +3,45 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 2.2.0 - TioDeNadal [2024-08-26]
```
Comment (Lucpen) [inline CHANGELOG.md:6]:
```suggestion
## 2.2.0 - TioDeNadal [2024-08-27]
```

**Pull request merged**